kmir/src/kmir/kdist/mir-semantics/rt/data.md

```diff
@@ -224,12 +227,9 @@ The `#setLocalValue` operation writes a `TypedLocal` value preceeding it in the
            .K
           ...
        </k>
-       <locals> LOCALS => LOCALS[I <- typedLocal(VAL, TY, mutabilityMut)] </locals>
-    requires 0 <=Int I
-     andBool I <Int size(LOCALS)
-     andBool isValue(VAL)
-     andBool tyOfLocal({LOCALS[I]}:>TypedLocal) ==K TY // matching type
-     andBool isMutable({LOCALS[I]}:>TypedLocal)        // mutable
+       <locals> _LOCALS[I <- typedLocal(_ => VAL, LOCALTY, mutabilityMut)] </locals>
+    requires isValue(VAL)
+     andBool LOCALTY ==K TY // matching type
```

Comment:
You can match on `VAL:Value` to omit `isValue(VAL)` here. You can also just give `LOCALTY` the name `TY` and omit `LOCALTY ==K TY` (it might make the equivalence easier to overlook, though).

Reply:
Thanks, I should indeed use `VAL:Value` here (as in the other places you pointed out).
As for `LOCALTY` vs `TY`, I have local changes to implement `Aggregate(..):Value`  that require the `TY` to become optional (the information is missing from the stable-mir JSON) so I'd rather keep this separate binding.